### PR TITLE
Fixes for rollback function.

### PR DESCRIPTION
**Pull request by pulltab**

### CODE REVIEW

File: src/gen_tw.erl [hunk 1]
```diff
@@ -184,25 +184,24 @@ loop(LVT, _Events=[#event{lvt=GVT, payload=?GVT_UPDATE_PAYLOAD}|T], PastEvents,
...
-%% First event in queue occurs before LVT.  Rollback to handle the event.
+%% First event in queue is an antievent for an event in PastEvents.  In this
+%% case we roll back to a state occuring before the antievent, and resume
+%% processing.  Placing events in PastEvents back into Events ensures that
+%% the antievent and event will cancel each other out in the Events queue.
+loop(LVT, Events=[#event{lvt=ELVT, not_anti=false}|_], PastEvents, Module, ModStates)
+        when ELVT =< LVT ->
+
+    NewModStates = lists:dropwhile(fun({SLVT, _}) -> SLVT >= ELVT end, ModStates),
+    [{LastKnownLVT, _}|_] = NewModStates,
```
Comment (mitchellwrosen) [inline src/gen_tw.erl]: Won't this pattern match fail if you receive an antievent for the earliest event?

Comment (pulltab) [inline src/gen_tw.erl]: We should never get an antievent at or before GVT.  The LVT we initialize at is assumed to be GVT.

Comment (pulltab) [inline src/gen_tw.erl]: There can be an explicit clause guarding against this if you would prefer.

At least exit with something more graceful that badmatch.

File: test/gen_tw_tests.erl [hunk 2]
```diff
@@ -180,11 +181,43 @@ rollback_skip_modstate(Pid) ->
...
+    timer:sleep(100),
+
+    %% Rollback to the middle
+    RollbackLVT = 49,
+    RBEvent1 = gen_tw:antievent(lists:nth(RollbackLVT, Events)),
+    gen_tw:notify(Pid, RBEvent1),
+
+    %% Ensure that it is safe to rollback to the beginning of time (GVT)
+    [FirstEvent|_] = Events,
+    gen_tw:notify(Pid, gen_tw:antievent(FirstEvent)),
```
Comment (pulltab) [inline test/gen_tw_tests.erl]: This is a case where we rollback to GVT.

File: src/gen_tw.erl (hunk 1 above)
Comment (pulltab) [inline src/gen_tw.erl]: See the new unit test for an example of rolling back to GVT.

**Pull request merged**